File: Year-2022/LeetCode-August-Challenge/FirstUniqueCharacterinaString.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    int firstUniqChar(string s) {
        int counts[26] = {0};
        for(auto ch : s) counts[ch - 'a']++;
        for(int i = 0; i < s.size(); ++i)
            if(counts[s[i] - 'a'] == 1) return i;
        return -1;
    }
};
int main(){
    Solution S;
    string s = "leetcode";
    cout<<S.firstUniqChar(s)<<endl;
    return 0;
}

File: Year-2022/LeetCode-August-Challenge/FirstUniqueCharacterinaString.js
var firstUniqChar = function(s) {
    let counts = [];
    counts.fill(0);
    for(let ch of s) counts[ch] ? counts[ch]++ : counts[ch] = 1;
    for(let i = 0; i < s.length; ++i)   
        if(counts[s[i]] === 1) return i;
    return -1;
};
let s = "leetcode";
console.log(firstUniqChar(s));